src/lxc/confile_oci.c

```diff
@@ -291,27 +318,208 @@ static int lxc_oci_linux_idmaps(json_t *elem, struct lxc_conf *conf, char type)
 
 static int lxc_oci_linux_sysctl(json_t *elem, struct lxc_conf *conf)
 {
-	WARN("The \"sysctl\" property is not implemented");
+	const char *key;
+	json_t *val;
+
+	json_object_foreach(elem, key, val) {
+		int ret;
+		char *param;
+
+		if (!json_is_string(val))
+			return -EINVAL;
+
+		ret = asprintf(&param, "lxc.sysctl.%s", key);
+		if (ret < 0)
+			return -EINVAL;
+
+		ret = set_config_sysctl(param, json_string_value(val), conf, NULL);
+		free(param);
+		if (ret < 0)
+			return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int lxc_oci_linux_namespaces(json_t *elem, struct lxc_conf *conf)
+{
+	size_t i;
+	json_t *it;
+
+	if (!json_is_array(elem))
+		return -EINVAL;
+
+	json_array_foreach(elem, i, it) {
+		int ret;
+		const char *type;
+		json_t *val;
+		if (!json_is_object(it))
+			return -EINVAL;
+
+		val = json_object_get(it, "type");
+		if (!json_is_string(val))
+			return -EINVAL;
+
+		type = json_string_value(val);
+		if (strcmp(type, "mount") == 0)
+			type = "mnt";
+		else if (strcmp(type, "network") == 0)
+			type = "net";
+		ret = set_config_namespace_clone("lxc.namespace.clone", type, conf, NULL);
+		if (ret < 0)
+			return ret;
+
+		val = json_object_get(it, "path");
+		if (!val)
+			continue;
+		if (!json_is_string(val))
+			return -EINVAL;
+		WARN("The namespaces \"path\" property is not implemented");
+	}
+
 	return 0;
 }
 
-static int lxc_oci_linux(json_t *elem, struct lxc_conf *conf)
+// https://github.com/opencontainers/runtime-spec/blob/v1.0.1/config-linux.md#default-devices
+// FIXME: use an absolute path for the destination (https://github.com/lxc/lxc/issues/2276)
+static const char * const default_devices[] = {
+	"/dev/null dev/null none bind,nosuid,noexec,create=file 0 0",
+	"/dev/zero dev/zero none bind,nosuid,noexec,create=file 0 0",
+	"/dev/full dev/full none bind,nosuid,noexec,create=file 0 0",
+	"/dev/random dev/random none bind,nosuid,noexec,create=file 0 0",
+	"/dev/urandom dev/urandom none bind,nosuid,noexec,create=file 0 0",
+	"/dev/tty dev/tty none bind,nosuid,noexec,create=file 0 0",
```

Comment:
You don't need this. If you simply set `lxc.autodev` to `1` below you get the exact same devices. Every runtime copied it from us. :)

Reply:
Actually, I think I do. With `autodev` I can't get the right options for this mount:
```
                {
                        "destination": "/dev",
                        "type": "tmpfs",
                        "source": "tmpfs",
                        "options": [
                                "nosuid",
                                "strictatime",
                                "mode=755",
                                "size=65536k"
                        ]
                },
```
So, I think I have to do everything manually.

Reply:
I mean sure you can do that but what's the problem with:

https://github.com/lxc/lxc/blob/a55e2ad107f084ba3b56fde234334f82ad184455/src/lxc/conf.c#L1200
```
ret = safe_mount("none", path, "tmpfs", 0, "size=500000, mode=755",
                 rootfs->path ? rootfs->mount : NULL);
```
the size? Please tell me that is not dictated by the spec.

Reply:
No it's not, it's dictated by the `options` from the runtime configuration, as shown above.
So, we need to match these options exactly. Even if I agree that I don't see where it would make a difference.